src/services/accommodation.service.js

```diff
@@ -0,0 +1,62 @@
+import db from '../database/models/index';
+
+const AccommodationService = {
+  async getAccommodationById(accommodationId) {
+    const accommodation = await db.Accommodations.findOne({
+      where: { id: accommodationId, },
+      include: [{ model: db.AccommodationImages, attributes: { exclude: ['id', 'accommodationid', 'createdAt', 'updatedAt'] } }],
+    });
+    return accommodation;
+  },
+
+  async getAllAccommodations() {
+    const accommodations = await db.Accommodations.findAll();
+    return accommodations;
+  },
+
+  async getAllAccommodationsByHost(hostId) {
+    const accommodations = await db.Accommodations.findAll({
+      where: { userid: hostId },
+      include: [{ model: db.AccommodationImages, attributes: { exclude: ['id', 'accommodationid', 'createdAt', 'updatedAt'] } }],
+    });
+    return accommodations;
+  },
+
+  async createAccommodation(request, userId) {
+    const {
+      name, description, address, availableSpace, cost, services, amenities
+    } = request;
+    const newAccommodation = await db.Accommodations.create({
+      name, description, address, availableSpace, cost, services, amenities, userid: userId
```

Comment:
you can replace all this line by : {...requst, userid: userId} and avoid destructuring the request variable